marlowe-playground-client/src/MainFrame/State.purs

```diff
@@ -322,101 +374,114 @@ handleAction s (HandleBlocklyMessage (CurrentCode code)) = do
   --   void $ query _blocklySlot unit (Blockly.SetError "You can't send new code to a running simulation. Please go to the Simulation tab and click \"reset\" first" unit)
   -- else do
   selectView Simulation
-  void $ toSimulation $ Simulation.handleAction s (ST.SetEditorText code)
+  void $ toSimulation $ Simulation.handleAction settings (ST.SetEditorText code)
+
+handleAction settings (HandleBlocklyMessage _) = do
+  -- Replicate the state of unsavedChanges from the submodule/subcomponent into the MainFrame state
+  hasUnsavedChanges <- queryCurrentEditorForUnsavedChanges
+  assign _hasUnsavedChanges hasUnsavedChanges
```

Comment:
Having information replicated will probably end up becoming a glitch at some point. Could we change the "getter" of the unsavedChanges in MainFrame to gather it from the various tabs instead of pushing it? I am aware the Hallogen signaling system is a bit messy, so maybe it is not worth it, I don't know

Reply:
This is one of my biggest pain points in this PR :/... For the submodules (JavaScript, Haskell, Marlowe) we can, as they have the state as part of the MainFrame state. For the subcomponents (Blockly, ActusBlockly) I think we can't, as we need to [query](https://pursuit.purescript.org/packages/purescript-halogen/5.0.0/docs/Halogen.Query.HalogenM#v:query) the child in the handleAction as it uses `HalogenM`. And the only way to communicate from the handleAction to the render is through the state.

@shmish111 please confirm

PD: I thought of a hacky way that includes wrapping the handleAction of the parent and childs to provide a mapping between states (so whenever a child changes it's states and force re-rendering of the child), we can project a part of the child state into the parent. But that may have other implications, so I don't think it's worth for this PR :/